compiler/lib/src/main/scala/analysis/Analysis.scala

```diff
@@ -70,7 +70,11 @@ case class Analysis(
   /** The dictionary under construction */
   dictionary: Option[Dictionary] = None,
   /** The telemetry packet set under construction */
-  tlmPacketSet: Option[TlmPacketSet] = None
+  tlmPacketSet: Option[TlmPacketSet] = None,
+  /** The dictionary generation mode */
+  dictionaryMode: Boolean = false,
```

Comment:
Let's rename this variable to be a predicate:
```suggestion
  /** Whether a dictionary is needed in code generation */
  dictionaryNeeded: Boolean = false,
```